dask/dataframe/shuffle.py

```diff
@@ -50,7 +50,10 @@ def _calculate_divisions(
         # 1) computing mins/maxes above, 2) every null being switched to NaN, and 3) NaN being a float.
         # Also, Pandas ExtensionDtypes may cause TypeErrors when dealing with special nulls such as pd.NaT or pd.NA.
         # If this happens, we hint the user about eliminating nulls beforehand.
-        if not is_numeric_dtype(partition_col.dtype):
+        if not (
+            is_numeric_dtype(partition_col.dtype)
+            or getattr(partition_col.dtype, "_is_numeric", False)
```

Comment:
I'd suggest moving this logic into a `is_numeric_dtype` method in `dask.dataframe._compat`. That way we can consolidate this extra workaround (which I expect we'll want to use other places where `pd.api.types.is_numeric_dtype` is currently used)